packages/core/src/render3/chained_injector.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injector} from '../di/injector';
+import {InjectFlags} from '../di/interface/injector';
+import {ProviderToken} from '../di/provider_token';
+import {NOT_FOUND_CHECK_ONLY_ELEMENT_INJECTOR} from '../view/provider_flags';
+
+/**
+ * Injector that looks up a value using a specific injector, before falling back to the module
+ * injector. Used primarily when creating components or embedded views dynamically.
+ */
+export class ChainedInjector implements Injector {
+  constructor(private injector: Injector, private moduleInjector: Injector) {}
```

Comment:
Can we rename `moduleInjector` -> `parentInjector`? From the `ChainedInjector` class perspective, there is no notion of module injector, it's just some other injector :)
```suggestion
  constructor(private injector: Injector, private parentInjector: Injector) {}
```